tests/bsim/bluetooth/host/att/sequential/dut/src/main.c

```diff
@@ -133,7 +133,7 @@ static void connect(void)
 	UNSET_FLAG(is_connected);
 
 	err = bt_le_scan_start(&scan_param, device_found);
-	ASSERT(!err, "Scanning failed to start (err %d)\n", err);
+	TEST_ASSERT(!err, "Scanning failed to start (err %d)\n", err);
```

Comment:
Might be outside the scope of this PR, but it's interesting how `bt_le_scan_stop` gets a `TEST_FAIL` but `bt_le_scan_start` gets a `TEST_ASSERT`

Reply:
Out of scope, but I agree that it should be investigated! :)